src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/SearchQueryOperation.java

```diff
@@ -87,27 +90,57 @@ public SearchResultView execute(final String view,
         return mergeDataToSearchResultOperation.execute(caseType.get(), searchResult, draftsAndCases, draftResultError);
     }
 
-    private SearchResult getSearchResultDefinition(final CaseTypeDefinition caseTypeDefinition, final String view) {
-        if (WORKBASKET.equalsIgnoreCase(view)) {
-            return uiDefinitionRepository.getWorkBasketResult(caseTypeDefinition.getId());
+    public SearchResult getSearchResultDefinition(final CaseTypeDefinition caseTypeDefinition, final String useCase) {
```

Comment:
I know it was like this already. But I don't think this method belongs here. The SearchQueryOperation should only concern with the actual execution of the external search. And be invoked only in that case. It's instead called before and after the actual search as well now which indicates it's doing too many unrelated things

Reply:
minor: I would be great if we can rename SearchResult to SearchResultDefinition